source/nz/net/ultraq/thymeleaf/fragments/FragmentMap.groovy

```diff
@@ -45,18 +45,43 @@ class FragmentMap extends HashMap<String,List<IModel>> {
 	/**
 	 * Set the fragment collection to contain whatever it initially had, plus the
 	 * given fragments, just for the scope of the current node.
+	 *
+	 * This must be used when processing non include/insert/replace related directives.
 	 * 
 	 * @param context
 	 * @param structureHandler
 	 * @param fragments The new fragments to add to the map.
 	 */
 	static void setForNode(IContext context, IElementModelStructureHandler structureHandler,
 		Map<String,List<IModel>> fragments) {
+		setForNodeInternal(context, structureHandler, fragments, false)
+	}
+
+	/**
+	 * Set the fragment collection to contain whatever it initially had, plus the
+	 * given fragments, just for the scope of the current node.
+	 *
+	 * This must be used when processing include/insert/replace related directives.
+	 *
+	 * @param context
+	 * @param structureHandler
+	 * @param fragments The new fragments to add to the map.
+	 */
+	static void setForNodeIncludeProcessing(IContext context, IElementModelStructureHandler structureHandler,
+		Map<String,List<IModel>> fragments) {
+		setForNodeInternal(context, structureHandler, fragments, true)
+	}
+
+	private static void setForNodeInternal(IContext context, IElementModelStructureHandler structureHandler,
+		Map<String,List<IModel>> fragments, boolean isIncludeProcessing) {
```

Comment:
In Groovy you can set default values for method parameters so you don't need to overload methods or make slightly different ones with additional flags here and there.  So this can all be simplified by adding `boolean isIncludeProcessing = false` as a new method parameter to the `setForNode` method:

```suggestion
		Map<String,List<IModel>> fragments, boolean isIncludeProcessing = false) {
```

The files that then call `setForNodeIncludeProcessing` can be put back to `setForNode`, but then add `true` as a 4th parameter to that call.